src/main/java/org/haffson/adventofcode/days/day05/Day05.java

```diff
@@ -0,0 +1,189 @@
+        return problemStatus;
+    }
+
+    @Nonnull
+    @Override
+    public String firstPart() {
+        final String fileName = "src/main/resources/puzzle_input/day5_input.txt";
+        return "Part 1 - Remaining Units after fully reacting the polymer: " + calculateRemainingUnits(fileReaders.getInputList(fileName));
+    }
+
+    @Nonnull
+    @Override
+    public String secondPart() {
+        final String fileName = "src/main/resources/puzzle_input/day5_input.txt";
+        return "Part 2 - Shortest remaining Units after fully reacting the polymer and removing one letter: " + calculateShortestRemainingUnits(fileReaders.getInputList(fileName));
+    }
+
+    /**
+     * Primary method for Day 5, Part 1.
+     * Calculates the remaining Units after fully reacting the polymer scanned.
+     *
+     * @return Int of remaining Units
+     */
+    private int calculateRemainingUnits(@Nonnull final List<String> inputStringList) {
+        String inputString = convertStringListToString(inputStringList);
+
+        List<Character> characterInputList = new ArrayList<>(convertStringToCharacterList(inputString));
+
+        List<Character> polymerCharacterList = removeCharDuplicatesWithDifferentCases(characterInputList);
+        return polymerCharacterList.size();
+    }
+
+    /**
+     * Helper method for Day 5, Part 1.
+     * Converts a List<String> to a single String.
+     * Needed because the FilerReader returns a List<String>.
+     *
+     * @return String of complete list
+     */
+    @Nonnull
+    String convertStringListToString(@Nonnull final List<String> inputStringList) {
+        return inputStringList.get(0);
+    }
+
+    /**
+     * Helper method for Day 5, Part 1.
+     * Converts a String to a List<Character>.
+     * Important: You get a mutable list.
+     *
+     * @return a mutable List<Character> for further functions.
+     */
+    @Nonnull
+    List<Character> convertStringToCharacterList(@Nonnull final String inputString) {
+        return inputString.chars().mapToObj(c -> (char) c).toList();
+    }
+
+    /**
+     * Helper method for Day 5, Part 1.
+     * Main method to remove duplicates.
+     * Not every duplicate should be removed.
+     * Just the duplicated adjacent values which have different cases (upper/ lower) will be removed.
+     * Loops over the list to remove emerging values too.
+     *
+     * @return List<Character> without adjacent duplicates.
+     */
+    @Nonnull
+    List<Character> removeCharDuplicatesWithDifferentCases(@Nonnull final List<Character> polymerCharacterList) {
+        boolean isRunning = true;
+        while (isRunning) {
+            isRunning = false;
+            for (int i = 0; i < polymerCharacterList.size() - 1; i++) {
+                Character actualCharacter = polymerCharacterList.get(i);
+                Character nextCharacter = polymerCharacterList.get(i + 1);
+                if (((Character.isUpperCase(actualCharacter) && Character.isLowerCase(nextCharacter))
```

Comment:
Strg-Alt-V um die if-condition in eine schön benannte Variable auszulagern (oder eine Funktion).

Reply:
Cool, den Trick kannte ich noch nicht